chipsec/helper/linux/helper.py

```diff
@@ -547,33 +547,34 @@ def read_mmio_reg(self, phys_address, size):
         reg = out_buf[:size]
         return defines.unpack1(reg, size)
 
-    def native_read_mmio_reg(self, phys_address, size):
+    def native_read_mmio_reg(self, bar_base, bar_size, offset, size):
+        if bar_size is None: bar_size = offset + size
         if self.devmem_available():
-            region = self.memory_mapping(phys_address, size)
+            region = self.memory_mapping(bar_base, bar_size)
             if not region:
-                os.lseek(self.dev_mem, phys_address, os.SEEK_SET)
-                reg = os.read(self.dev_mem, size)
-            else:
-                region.seek(phys_address - region.start)
-                reg = region.read(size)
+                self.native_map_io_space(bar_base, bar_size)
+                region = self.memory_mapping(bar_base, bar_size)
```

Comment:
Couple of questions:
Have you tested cases where bar_size is None and you access the region (updating mapping) multiple times?  An example would be reading offset 0 and then offset 0x10.
Are we always able to map a region or do we need some error handling in case after mapping the IO space in case memory_mapping returns None?

Reply:
I have tested where bar_size is None and did not encounter any problems.  The dump command does this by default.  
Added an error message if the region does not map.